### ISSUE TITLE
Migrate data structure definitions to Hydra

### ISSUE THREAD

**Issue by joshsh** [synchrony/smsn#63]
Currently in SmSn, atoms and other data structures are specified by hand-coded Java classes. In order to make SmSn graphs accessible to Hydra, and accessible in multiple programming languages (particularly Haskell as well as Java), we need to create a SmSn data model in Hydra. This will have the side effect of cleanly separating the data model from behaviors associated with the data model.

See in particular `net.fortytwo.smsn.brain.model`. Additionally, migrating the Action classes (`net.fortytwo.smsn.server.actions`) will simplify any future client/server IO.
 
cc @jmatsushita

**Comment by jmatsushita** [comment]
> migrating the Action classes
I suppose you mean, migrating them to programs written in the Hydra DSL?

**Comment by joshsh** [comment]
Correct. These DSL-based model specifications will be placed in the smsn-models module (created last fall, but not yet populated).
